### ISSUE TITLE
Attempting conversion containing `for_each` with primitives fails

### ISSUE THREAD

**Issue by smithrobs** [pulumi/pulumi-converter-terraform#228]
### What happened?

The `pulumi convert --from terraform --language python` command does not work for terraform for-loops that have primitives. For example `for_each = var.enable ? [1]  : []` which we typically do for dynamic blocks such as S3 lifecycle rules which has to be defined in a single resource ([ref](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_lifecycle_configuration)). The error message is:
```
error: main.pp:7,40-75: cannot traverse value of type number;
error: main.pp:7,40-75: cannot traverse value of type union(cty.NumberIntVal(1), none);
error: could not generate output program
```

### Example

```hcl
resource "aws_s3_bucket_lifecycle_configuration" "this" {
  count  = length(var.expirations) > 0 || length(var.transitions) > 0 || length(var.noncurrent_version_transitions) > 0 ? 1 : 0
  bucket = aws_s3_bucket.this.id
  
  # There are some other dynamic rule blocks i have omitted

  dynamic "rule" {
    for_each = var.transitions
    content {
      status = try(rule.value.enabled, true) ? "Enabled" : "Disabled"
      id     = try(rule.value.id, md5("transitions${jsonencode(rule.value)}"))
      dynamic "abort_incomplete_multipart_upload" {
        # following line will fail to convert
        for_each = try(rule.value.abort_incomplete_multipart_upload_days, null) != null ? [1] : []
        content {
          days_after_initiation = rule.value.abort_incomplete_multipart_upload_days
        }
      }
      filter {
        prefix = try(rule.value.prefix, "")
        dynamic "tag" {
          for_each = try(jsondecode(rule.value.tags), [])
          content {
            key   = try(tag.key, null)
            value = try(tag.value, null)
          }
        }
      }
      transition {
        date          = try(rule.value.date, null)
        days          = try(rule.value.days, null)
        storage_class = rule.value.storage_class
      }
    }
  }
}
```

### Output of `pulumi about`

Not available; assume v3.141.0

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by joeduffy** [comment]
@brandonpollack23 Would love it if you could take a look at this. Sounds like it could be impactful for the customer @smithrobs is working with!

**Comment by brandonpollack23** [comment]
Taking a look!

**Comment by brandonpollack23** [comment]
Alright so just a heads up on a few things:

If the variables are not defined in the TF (I assume they are and just ommitted here) then that causes some trouble since our converter doesnt know what the references are to.

Another issue is we've done a bunch of work getting conversions more robust recently but `try` remains particularly difficult.  We have a rudimentary version in pulumi already, but the convert doesn't emit it because there are quite a few cases where it breaks things.

Let me try to see how it handles this case and then I'll see if there are any other things we need to do to get this working and I'll report back shortly

**Comment by brandonpollack23** [comment]
Alright so the good news is the "try" we have implemented seems to cover this case pretty well (usually this kind of count is 0 or 1 pattern works fine, it's when you need to do .apply that we get hairy).

But there is some kind of typing error on the variable so Ill spend a bit more time on it and see if its a quick fix

**Comment by brandonpollack23** [comment]
Ok after mulling on this a while I have a hunch:

I think TF supports "iterating" over scalars as singular lists and Pulumi doesn't right now, let me double check this and see if that is the case and if I can fix it

**Comment by brandonpollack23** [comment]
So to summarize:

1. Try is not yet released, once it is that is the first step: pulumi/pulumi#18350, enabled on the converter by #295
2. Once that is done, the bug found by this needs to be fixed: #18718
3. After working around this by editing the pcl #18649 still seems to occur.

Once these are all fixed this should work.

To give you a little timeline: I'm looking at (2) now, (1) is my next major task and hopefully we have it done in a few weeks.  (3) can probably be mixed in between (or will be fixed incidentally by the continuing work).

**Comment by brandonpollack23** [comment]
Just a quick update, the shadow overlap issue has been fixed #18718, but the other work that will fully unblock this (dealing with try) is still underway.  We found some other slight issues with our internal binding that we need to address, but I'm putting out the initial implementaiton for nodejs today and python early next week.  We're hoping this will be fully unblocked soon
